### PR TITLE
Hide Warnings on dropped rows in the import-issues panel

### PR DESCRIPTION
**Pull request by StepKie**
## Summary

User flagged: rows showing up twice in the import-issues panel — once under a Warning group, once under an Error group. Reproduces on `moxfield-collection.csv`: `SetInfoEnricher` emits *"Set code 'MB1' not found in Scryfall data"* (Warning), then `CatalogValidator` immediately drops the row with *"No printing at MB1 #62 in Scryfall data"* (Error). Both are true facts about the same row, but visually they read as redundant.

**Fix is render-only:** when a row has an Error, hide its Warnings. The Error already names the failure precisely; the precondition Warning is noise once the row is dropped.

| | |
| --- | --- |
| `MtgCsvProcessor.razor` | Build a `HashSet<int>` of row numbers with Errors. Filter `displayedIssues` = Errors + Warnings whose row isn't in that set. The grouping below uses `displayedIssues` instead of `r.Issues`. |
| Top-of-panel chips | Switched to row-based counts (distinct rows with errors / warnings) instead of issue-based. Previously a row with two issues double-counted. |
| Underlying `r.Issues` | Unchanged. Error-report CSV download + the Report-to-GitHub payload remain exhaustive on purpose — those audiences want the full enricher trail. UI is the only layer where the overlap reads as redundancy. |

Plus a small SetInfoEnricher diagnostic test (`EnrichAsync_UnknownSetCode_WithCsvProvidedSetName_DoesNotWarn`) — comes out of the diagnosis trail for this fix. Pins the existing behavior on the corner case where a CSV supplies both an unknown set code AND a CSV-provided set name (no warning fires for that combination).

## Test plan

- [x] `dotnet test MtgCsvHelper.slnx` — 175 passing
- [x] Local dev-server verification with `moxfield-collection.csv`: rows with the MB1 set-code Warning + the resulting "No printing at MB1 #N" Error now show only the Error group entry; warning chip count drops accordingly ✅
- [ ] CI green

### CODE REVIEW

File: MtgCsvHelper.Tests/SetInfoEnricherTests.cs [hunk 1]
```diff
@@ -38,6 +38,32 @@ public async Task EnrichAsync_NonCanonicalSetName_OverwrittenByCatalog()
 	// (Moxfield / Manabox / Archidekt / TopDecked all emit canonical names), the catalog
 	// overwrite is a no-op. Documents the intended boundary so the "catalog always wins" rule
 	// can't silently degrade into "catalog overwrites with garbage when name lookup misfires".
+	// Diagnostic — pinning the current behavior for an "MB1 + Mystery Booster" row whose
+	// set code is unknown to the catalog. Reported by user: both SetInfoEnricher emits a
+	// warning ("Set code 'MB1' not found") AND CatalogValidator emits an error
+	// ("No printing at MB1 #N"). One issue per row, not two, is the expected UX.
+	[Fact]
+	public async Task EnrichAsync_UnknownSetCode_WithCsvProvidedSetName_DoesNotWarn()
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper.Tests/SetInfoEnricherTests.cs:67]: ✅ Done in 1ab0a81 — better than you flagged actually. The orphaned docblock at lines 37-40 was the *previous* test's ("Companion to the test above…") and my new test got inserted between it and the test it described. Moved the new test to the bottom of the class so the original docblock re-pairs with `EnrichAsync_AlreadyCanonicalSetName_PassesThroughUnchanged`. Behaviour unchanged; file reads coherently now.
